stock_inventory_barcode/wizard/stock_inventory_barcode.py

```diff
@@ -0,0 +1,284 @@
+        res = {"warning": {}}
+        if self.product_id and self.location_id and self.uom_id:
+            if self.uom_id != self.product_id.uom_id:
+                self.uom_id = self.product_id.uom_id
+                res["warning"] = {
+                    "title": _("Error"),
+                    "message": _(
+                        "You cannot change the unit of measure of the "
+                        "product: it is been restored to '%s'."
+                    )
+                    % self.product_id.uom_id.name,
+                }
+            if not self.product_id.tracking or self.product_id.tracking == "none":
+                if self.lot_id:
+                    raise UserError(
+                        _(
+                            "Product '%s' is not tracked by lot/serial so the "
+                            "lot field must be empty. This should never "
+                            "happen."
+                        )
+                        % self.product_id.display_name
+                    )
+                self.update_wiz_screen(res)
+            elif self.lot_id:
+                self.update_wiz_screen(res)
+        return res
+
+    def update_wiz_screen(self, res):
+        silo = self.env["stock.inventory.line"]
+        ilines = silo.search(
+            [
+                ("inventory_id", "=", self.inventory_id.id),
+                ("product_id", "=", self.product_id.id),
+                ("location_id", "=", self.location_id.id),
+                ("prod_lot_id", "=", self.lot_id and self.lot_id.id or False),
+            ]
+        )
+        if len(ilines) == 1:
+            self.inventory_line_id = ilines[0]
+            self.change_qty = ilines[0].product_qty
+        elif len(ilines) > 1:
+            res["warning"] = {
+                "title": _("Error"),
+                "message": _(
+                    "Several inventory lines exists for this product "
+                    "(and lot) on the same stock location. "
+                    "This should happen only when using packaging, "
+                    "but this scenario is not supported for the moment."
+                ),
+            }
+        else:
+            new_iline = silo.create(
+                {
+                    "inventory_id": self.inventory_id.id,
+                    "product_id": self.product_id.id,
+                    "prod_lot_id": self.lot_id and self.lot_id.id or False,
+                    "product_uom_id": self.uom_id.id,
+                    "location_id": self.location_id.id,
+                }
+            )
+            self.inventory_line_id = new_iline
+        domain_same_product = [
+            ("inventory_id", "=", self.inventory_id.id),
+            ("product_id", "=", self.product_id.id),
+            ("product_qty", ">", 0),
+        ]
+        if ilines:
+            domain_same_product += [("id", "not in", ilines.ids)]
+        same_product_lines = silo.search(domain_same_product)
```

Comment:
so I was trying to trigger notes and couldn't.

the section about managing identical product lines seems obsolete because of:

https://github.com/odoo/odoo/blob/9b23864027bc24032993833f213395cb45fe86bb/addons/stock/models/stock_inventory.py#L477